docs/reference/settings/transform-settings.asciidoc

```diff
@@ -17,13 +17,18 @@ default.
 `node.roles: [ transform ]`::
 (<<static-cluster-setting,Static>>) Set `node.roles` to contain `transform` to
 identify the node as a _transform node_. If you use the default settings for a
-node, it does not have the `transform` role.
+node, it does not have the `transform` role. However, by default all generic
+data nodes are also {transform} nodes.
```

Comment:
This is now very complicated.  I don't think you should use the phrase "generic data nodes", because what does that mean?

If you specify `node.data: true`, don't specify `node.roles` at all and don't specify `node.transform` at all then the node is a transform node.  (`node.data` and `node.transform` are old deprecated settings that will not work at all in 8.x.  So the statement used to make sense before `node.roles` was introduced.)

If you specify `node.roles: [ data ]` then the node is _not_ a transform node.

If you don't specify any node role settings (old or new) then the node is both a data node and a transform node (and every other type of node).

A data/transform node that specifies `node.roles` needs to explicitly say both, for example, `node.roles: [ data, transform ]` or `node.roles: [ data_hot, transform, remote_cluster_client ]`.

Reply:
Thanks for the explanation!  I've removed "All data nodes are also transform nodes" from the node settings page and re-worked the details in the transform and ML settings pages to try to make this clearer.